benchmarks/CallDepth.bench.cpp

```diff
@@ -0,0 +1,204 @@
+        if (call_idx == 0) {
+            co_return co_await service1(filename, n, e, m);
+        } else {
+            co_return co_await service3(filename, n, e, m);
+        }
+    }
+
+    Lazy<uint64_t> service3(const char *filename, int n, IOExecutor *e, int m) {
+        n--;
+        if (n == 0) {
+            co_return co_await read_db(filename, e, m);
+        }
+        int call_idx = n % 2;
+        if (call_idx == 0) {
+            co_return co_await service1(filename, n, e, m);
+        } else {
+            co_return co_await service2(filename, n, e, m);
+        }
+    }
+
+    Lazy<uint64_t> read_db(const char *filename, IOExecutor *e, int m) {
+        std::vector<Lazy<std::size_t>> lazy_list;
+        for (int i = 0; i < m; i++) {
+            lazy_list.push_back(async_read_file(filename, e));
+        }
+        [[maybe_unused]] auto ret =
+            co_await collectAllPara(std::move(lazy_list));
+        co_return 0;
+    }
+};
+
+void Uthread_call_depth_bench(benchmark::State &state) {
+    using namespace async_simple::uthread;
+    int n = state.range(0);
+    std::string s = std::string("test.") + std::to_string(4) + ".txt";
+    gen_file(s, 4);
+    auto core_num = std::thread::hardware_concurrency();
+    SimpleExecutor e(core_num);
+    for (auto _ : state) {
+        state.PauseTiming();
+        std::vector<std::function<void()>> task_list;
+        task_list.reserve(task_num);
+        std::atomic<bool> running = true;
+        int m = 1;
+        for (int i = 0; i < task_num; ++i) {
+            task_list.emplace_back([n, m, s, e = &e]() {
+                Service service{
+                    .n = n, .m = m, .filename_ = s.c_str(), .e_ = e};
+                service.run();
+            });
+        }
+        state.ResumeTiming();
+        async<Launch::Schedule>(
+            [&running, &e, task_list]() mutable {
+                collectAll<Launch::Schedule>(task_list.begin(), task_list.end(),
+                                             &e);
+                running = false;
+            },
+            &e);
+        while (running) {
+        }
+    }
+}
+
+void Lazy_call_depth_bench(benchmark::State &state) {
+    using namespace async_simple::coro;
+    int n = state.range(0);
+    auto file_size = 4;
+    std::string s = std::string("test.") + std::to_string(file_size) + ".txt";
+    gen_file(s, file_size);
+    auto core_num = std::thread::hardware_concurrency();
+    SimpleExecutor e(core_num);
+    for (auto _ : state) {
+        auto f = [&s, n, e = &e]() mutable -> Lazy<void> {
+            std::vector<Lazy<uint64_t>> lazy_list;
+            lazy_list.reserve(task_num);
+            int m = 1;
+            for (int i = 0; i < task_num; ++i) {
+                lazy_list.push_back(
+                    AsyncService().run(s.c_str(), n, e->getIOExecutor(), m));
+            }
+            [[maybe_unused]] auto ret =
+                co_await collectAllPara(std::move(lazy_list));
```

Comment:
also should not include task schedule latency, collectAllPara will occurred the task schedule.